### PR TITLE
modify the way of RPC request and response & add RPC some interface & add RPC Demo

### PR DESCRIPTION
**Pull request by fqliao**

### CODE REVIEW

File: librpc/Rpc.cpp [hunk 1]
```diff
@@ -22,579 +22,644 @@
...
-            Json::Value peerJson;
-            peerJson["NodeID"] = peer.nodeID.hex();
-            peerJson["IP"] = peer.nodeIPEndpoint.address.to_string();
-            peerJson["TcpPort"] = peer.nodeIPEndpoint.tcpPort;
-            peerJson["UdpPort"] = peer.nodeIPEndpoint.udpPort;
-            peerJson["Topic"] = Json::Value(Json::arrayValue);
-            for (std::string topic : peer.topics)
-                peerJson["Topic"].append(topic);
-            responseJson["result"].append(peerJson);
+            auto status = sync->status();
```
Comment (wheatli) [inline librpc/Rpc.cpp]: Let the sync->status() return a string, and the RPC api here just gets status and packs string as response

File: librpc/Rpc.cpp [hunk 2]
```diff
@@ -22,579 +22,644 @@
...
+                    {
+                        response["transactions"].append(toJson(
+                            transactions[i], std::make_pair(hash, i), block->header().number()));
+                    }
+                    else
+                    {
+                        response["transactions"].append(toJS(transactions[i].sha3()));
+                    }
+                }
+            }
```
Comment (wheatli) [inline librpc/Rpc.cpp]: you should return some error msg to client, indicate what  happen

File: test/unittests/libp2p/FakeHost.h [hunk 3]
```diff
@@ -163,18 +163,38 @@ class FakeSessionFactory : public SessionFactory
...
+
 /// Fakes of Host
 class FakeHost : public Host
 {
 public:
     FakeHost(std::string const& _clientVersion, KeyPair const& _alias, NetworkConfig const& _n,
         std::shared_ptr<AsioInterface>& m_asioInterface, shared_ptr<SocketFactory>& _socketFactory,
-        shared_ptr<SessionFactory>& _sessionFactory, shared_ptr<ba::ssl::context> _sslContext)
+        shared_ptr<SessionFactory>& _sessionFactory, shared_ptr<ba::ssl::context> _sslContext,
+        bool forRpc = false)
```
Comment (wheatli) [inline test/unittests/libp2p/FakeHost.h]: Does it support for RPC now ?

File: librpc/Rpc.cpp (hunk 1 above)
Comment (fqliao) [reply librpc/Rpc.cpp]: Ok, I will solve it with jimmyshi.

File: librpc/Rpc.cpp (hunk 2 above)
Comment (fqliao) [reply librpc/Rpc.cpp]: I will fix it.

File: test/unittests/libp2p/FakeHost.h (hunk 3 above)
Comment (fqliao) [reply test/unittests/libp2p/FakeHost.h]: It supports RPC and P2P UT

**Pull request merged**